tendermint/src/lite_impl/signed_header.rs

```diff
@@ -16,15 +16,11 @@ impl lite::Commit for block::signed_header::SignedHeader {
         // NOTE we don't know the validators that committed this block,
         // so we have to check for each vote if its validator is already known.
         let mut signed_power = 0u64;
-        for vote_opt in &self.iter() {
+        for vote in &self.iter().unwrap() {
```

Comment:
Can't this panic? Why are there no `None` in the `Option<vote::SignedVote>`s returned by the iter? Ah OK, I see the method signature changed to ` fn iter(&self) -> Result<Vec<vote::SignedVote>, Error>`. Why is the error not handled?

Reply:
Based on the comment right under it, I guess it can. Maybe iter() should be iterated and unwrapped with check in the loop.

Something like
```
for vote in $self.iter() {
  if vote.is_none() {
    continue
  }
  unwrapped_vote = vote.unwrap()
..
``` ?

Reply:
We shouldn't use `unwrap` here but handle the error instead (this could mean to just bubble it up using the `?` operator via `&self.iter()?`; depends what we want to do with it but this would at least not panic)

Reply:
Looks like this error can only occur if there is a `CommitSig` with a missing (`None`), `validator_address`. If the address is in fact not optional, we could get rid of the error altogether.

Reply:
The address is optional if the vote is Absent, ie. we have BlockIDFlagAbsent

Reply:
@Liamsi You're right, the error occurs only when there is a missing `validator_address` with a non-absent block ID flag i.e. not `BlockIDFlagAbsent`. Should we simply `continue` if that happens or return an error from within `voting_power_in` ?

Reply:
Didn't think this through but sounds like an error to me. But I'd double check with the spec and maybe how the current go code deals with it. Can think about it more tmrw.

Reply:
I think if this is not `BlockIDFlagAbsent` it is an `error` if the `validator_address` is missing (and we should not simply continue).

Reply:
Thinking about this further, maybe we should rename that method (iter) to be more explicit about what it does (it does some validation, extracts and returns valid SignedVotes from the SignedHeader, or, returns an error). Keeping the current name and flow is quite unidiomatic: this method doesn't implement an iterator (anymore).

Reply:
It's very unlike that we receive a commit with a missing `validator_address` and yeah i guess this can be an `ImplementationSpecific` error? 

Hmm, the method converts `CommitSigs` to `SignedVotes` with some validation done there  and returns a `Vec<SignedVote>` (when Result is Ok). Should we call it something like `signed_votes_iter`?

Reply:
What would e best: there is already a `validate` method that iterates over the commit's signatures. We can extend that loop to check for one sig with `!commit_sig.is_absent()` there is a `None` `commit_sig.validator_address` (the same validation that currently happens in `commit_to_votes` / `iter`).

We can rename `iter` to `signed_votes`.  We can collapse `commit_to_votes` and make it return `Vec<vote::Vote>` and not a result.

Reply:
And yes, the error would be `ImplementationSepcific`

Reply:
I'd like to refactor this in a PR against this branch if that's OK for everyone.

Reply:
Here's the PR for that: #199 (https://github.com/informalsystems/tendermint-rs/pull/199)